packages/ui/react-ui-table/src/theme.ts

```diff
@@ -2,66 +2,98 @@
 // Copyright 2023 DXOS.org
 //
 
-import { chromeSurface } from '@dxos/react-ui-theme';
-
-// TODO(burdon): Remove nested classNames? Add to theme?
-export type TableSlots = {
-  root?: {
-    className?: string | string[];
-  };
-  table?: {
-    className?: string | string[];
-  };
-  header?: {
-    className?: string | string[];
-  };
-  footer?: {
-    className?: string | string[];
-  };
-  group?: {
-    className?: string | string[];
-  };
-  row?: {
-    className?: string | string[];
-  };
-  cell?: {
-    className?: string | string[];
-  };
-  focus?: {
-    className?: string | string[];
-  };
-  selected?: {
-    className?: string | string[];
-  };
-  margin?: {
-    className?: string | string[];
-  };
-};
-
-// TODO(burdon): Change to groups.
-/*
-head: {
-  default:
-  cell:
-}
-row: {
-  default: '',
-  selected: ''
-}
-*/
-
-// TODO(burdon): Scrollbar area.
-// TODO(burdon): Overscroll horizontal (full width).
-
-// TODO(burdon): Integrate with DXOS UI theme (direct dependency -- see react-ui-editor, tailwind.ts).
-//  See Link.tsx const { tx } = useThemeContext();
-//  Reuse button fragments for hoverColors, selected, primary, etc.
-export const defaultTableSlots: TableSlots = {
-  // TODO(burdon): head/body/table rows.
-  header: { className: [chromeSurface, 'px-2 font-light select-none'] },
-  footer: { className: [chromeSurface, 'px-2 font-light'] },
-  cell: { className: 'px-2' },
-  group: { className: 'px-2 font-light text-xs text-left' },
-  focus: { className: 'ring ring-primary-600 ring-inset' },
-  selected: { className: '!bg-teal-100 dark:!bg-teal-700' },
-};
+import { type CellContext } from '@tanstack/react-table';
+
+import { chromeSurface, fixedSurface, focusRing, groupBorder, hoverColors, mx } from '@dxos/react-ui-theme';
+import { type ComponentFunction } from '@dxos/react-ui-types';
+
+import { type TableContextValue, type TableFlags } from './components';
+
+export const currentRow = '!bg-neutral-75 !dark:bg-neutral-850';
+export const selectedRow = '!bg-primary-100 dark:!bg-primary-700';
```

Comment:
promote fragments to main theme fragments for reuse?

Reply:
I think I’d actually like to do this as a follow-up, I know you need these too but I’d like to do a bit of design planning, then share the result in a quick iteration loop.